app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,61 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]
+
+  def index
+    @posts = Post.all.order('created_at DESC')
+    @post = Post.new
+  end
+
+  def show; end
+
+  def new
+    @post = current_user.posts.build
```

Comment:
Consider using `Post.new`

Reply:
Hello and thanks Maya for your wonderful review. I did implement this project twice, one with the device gem and the other from scratch. I have been told to use the devise gem again and that is why I am resubmitting this one for a code review again after fixing the minor issues you indicated. I am also confused because Odin say one should use the device gem.